### PR TITLE
UG and DG Enhancements and Fix Find Command

### LINKED ISSUE #78: [PE-D][Tester A] 'find' inputting format differs from the examples.
similar issue with issue #2 and #4, the inputting format shows a space between / and KEYWORD as 'find FIELD/ KEYWORD FIELD/ KEYWORD', but the examples does not have any space between / and the KEYWORD as 'find n/John t/friend'. This could be confusing for the readers, try to make the inputting format and examples the same. 

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: doyelee0313/ped#5

### LINKED ISSUE #88: [PE-D][Tester D] Misleading Search Command Format
Search command general format says:

find FIELD/ KEYWORD FIELD/ KEYWORD...

This implies that there must be at least 2 FIELD and KEYWORD pairs even though that is not a requirement. You should put the second pair in square brackets, like so:
find FIELD/ KEYWORD [FIELD/ KEYWORD...]

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: vannykin/ped#4

### LINKED ISSUE #91: [PE-D][Tester C] Should be able to `find` by email
The `find` command in the current version does not support searching by email address, a critical oversight that diminishes its utility for users who rely on email as a key identifier.

This absence forces users to use less efficient methods for locating specific entries, leading to increased search times and potential frustration. As email search is a standard feature in similar data retrieval tools, its absence in the find command signifies a notable deficiency in the product design, making the feature incomplete for practical usage.

 

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: minreiseah/ped#7

### LINKED ISSUE #92: [PE-D][Tester D] Page 5 Bullet Points
Awkward use of bullet points on page 5 (Search Guidelines).

One example should not be separated into 3 different bullet points!

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: vannykin/ped#6

### LINKED ISSUE #94: [PE-D][Tester D] Awkward Spacing on Page 5
Weird spacing on page 5 (search guidelines).

Main bullet point and sub bullet points are far apart, but sub bullet points are close to the next main bullet point that they do not belong to.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: vannykin/ped#8

### LINKED ISSUE #96: [PE-D][Tester B] Verbose explanation of already explained or mostly intuitive things in `find`
The documentation for find is way too wordy.

It has an explanation of empty tags which is inconsistent from everywhere else. It should be consistent and explained in every command if necessary, but the program's error message already handles that.

"Logical and" and case sensitivity was already explained before this section.

The examples are not really adding anything other than maybe the case-sensitivity one. However, that could have just been explained in simpler terms.

I think this section could be removed and the unique info it does have can be moved to the preceding section.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: miffi/ped#3

### LINKED ISSUE #98: [PE-D][Tester D] Typo on Page 6
Typo: 'sorted ** first by date from...'

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: vannykin/ped#9

### LINKED ISSUE #113: [PE-D][Tester C] The find command should allow searching by phone number
Searching by phone number is an important feature that every contact management system should have. This critical oversight that diminishes its utility for users who rely on phone numbers as a key identifier.

This absence forces users to use less efficient methods for locating specific entries, leading to increased search times and potential frustration. As number search is a standard feature in similar data retrieval tools, its absence in the find command signifies a notable deficiency in the product design, making the feature incomplete for practical usage.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: minreiseah/ped#15

### LINKED ISSUE #121: [PE-D][Tester A] Spelling mistake in 'find' description
There is a typo in the 'find' description.

"Teh search will find contacts containing the provided keyword" "Teh" should be "The".

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: doyelee0313/ped#12

### LINKED ISSUE #126: [PE-D][Tester B] Typos
-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: miffi/ped#11

### LINKED ISSUE #128: [PE-D][Tester B] find doesn't work as specified in error output
Find shows that the parameters should be in this format

But this is not the same as defined in the feature guide.

If a person were to follow this, there's a high chance the command will not do what expected.

For example, in the sample data, running `find n/ber nice` returns 0 persons when there is "Bernice Yu" in the data

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: miffi/ped#13

### PR DESCRIPTION
**Pull request by maze508**
- Resolves #128 
- Resolves #126 
- Resolves #121 
- Resolves #113 
- Resolves #98 
- Resolves #96 
- Resolves #94 
- Resolves #92 
- Resolves #91 
- Resolves #88 
- Resolves #78

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -561,3 +561,11 @@ testers are expected to do more _exploratory_ testing.
...
+
+4. The current implementation of the find command allows users to search for contacts based on their names, tags, or company names. However, it does not support searching by address, email, or phone number. We acknowledge that the ability to search by these fields can significantly enhance user experience by providing more flexibility and efficiency in locating contact information. The initial decision to exclude address, email, and phone number from the search criteria was based on a focus on the most commonly used identifiers for quick search and to maintain simplicity in the search interface. We also considered the privacy implications and the less frequent necessity of searching by personal information such as phone numbers or addresses. However, in order to enhance the utility of our contact management system, we are planning to introduce expanded search capabilities. This will include the ability to search for contacts by their phone numbers, email addresses, and physical addresses. This enhancement aims to provide a comprehensive search functionality that meets the needs of all users, making the tool more versatile and efficient for locating specific entries.
```
Comment (guanquann) [inline docs/DeveloperGuide.md:571]: I think it would be better if you follow the format given in Week 12 Project section

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 2]
```diff
@@ -18,9 +18,13 @@ public class FindCommand extends Command {
...
-            + "Parameters: n/KEYWORD [MORE_KEYWORDS]... t/KEYWORD [MORE_KEYWORDS]... c/KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " n/alice t/friends c/Meat \n"
-            + "Note: Multiple keywords (name, tag or company) are treated with a logical AND.";
+            + "Parameters: [n/NAME_KEYWORDS] [t/TAG_KEYWORDS] [c/COMPANY_KEYWORDS]\n"
+            + "Each field is optional, "
+            + "but at least one must be provided. All Keywords are combined using logical AND (&&).\n"
+            + "Example: " + COMMAND_WORD + " n/alice t/friends c/Meat - "
+            + "Finds all persons named 'alice' AND tagged as 'friends' AND who's company name is "
+            + "associated with 'Meat'.\n";
+
```
[the remark this answers is not in the document]
Comment (gavingoh99) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:27]: I believe this is just an update to documentation - no change made to the actual command

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -583,3 +583,6 @@ to view their newly added order. We plan to show a preview of the order added. F
...
...h Functionality**: The current implementation of the find command allows users to search for contacts based on their names, tags, or company names. However, it does not support searching by address, email, or phone number. We acknowledge that the ability to search by these fields can significantly enhance user experience by providing more flexibility and efficiency in locating contact information. The initial decision to exclude address, email, and phone number from the search criteria was based on a focus on the most commonly used identifiers for quick search and to maintain simplicity in the search interface. We also considered the privacy implications and the less frequent necessity of searching by personal information such as phone numbers or addresses. However, in order to enhance the utility of our contact management system, we are planning to introduce expanded search capabilities. This will include the ability to search for contacts by their phone numbers, email addresses, and physical addresses. This enhancement aims to provide a comprehensive search functionality that meets the needs of all users, making the tool more versatile and efficient for locating specific entries.
```
Comment (guanquann) [inline docs/DeveloperGuide.md:587]: Should bold the `:` as well 😆

File: docs/UserGuide.md [hunk 4]
```diff
@@ -235,46 +235,50 @@ Examples:
 - Search feature supports substring search by name and/or tags and/or company **ONLY**.
 - Finds all contacts whose names, tags or company matches the substring keyword provided.
 
-
-General Format: `find FIELD/ KEYWORD FIELD/ KEYWORD ...`
+General Format: `find [FIELD/KEYWORD] [FIELD/KEYWORD] ...`
```
Comment (guanquann) [inline docs/UserGuide.md:238]: Should the first `[FIELD/KEYWORD]` be compulsory?

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (maze508) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:27]: https://github.com/nus-cs2103-AY2223S2/forum/issues/310
